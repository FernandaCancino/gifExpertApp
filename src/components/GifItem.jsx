import PropTypes from 'prop-types'

export const GifItem = (image) => {
  return (
    <div className="card">
      <img src={image.url} alt={image.title} />
      <p>{image.title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}