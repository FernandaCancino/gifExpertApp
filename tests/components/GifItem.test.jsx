import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('pruebas en el <GifItem>', () => {

  const title='Jujutsu';
  const url= 'https://jujutsu.org/';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el url y el alt indicado', () =>{
    render(<GifItem title={title} url={url} />);
    // en vez de hacer todo esto
    // expect( screen.getByRole('img').src ).toBe( url );
    // expect( screen.getByRole('img').alt ).toBe( title );

    //se puede hacer asi
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url ); 
    expect( alt ).toBe( alt ); 
  });

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect( screen.getByText(title) ).toBeTruthy();
  });
})