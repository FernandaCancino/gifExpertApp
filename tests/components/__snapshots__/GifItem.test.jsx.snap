// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <GifItem> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Jujutsu"
      src="https://jujutsu.org/"
    />
    <p>
      Jujutsu
    </p>
  </div>
</div>
`;
